python/src/linalg.cpp

```diff
@@ -0,0 +1,218 @@
+                        a,
+                        "-inf",
+                        get_reduce_axes(axis, a.ndim()),
+                        keepdims,
+                        stream);
+                  }
+                  return norm(
+                      a,
+                      p,
+                      std::holds_alternative<std::monostate>(axis)
+                          ? std::vector<int>()
+                          : get_reduce_axes(axis, a.ndim()),
+                      keepdims,
+                      stream);
+                },
+                [&](const std::string& p) {
+                  return norm(
+                      a,
+                      p,
+                      std::holds_alternative<std::monostate>(axis)
+                          ? std::vector<int>()
+                          : get_reduce_axes(axis, a.ndim()),
+                      keepdims,
+                      stream);
+                },
+                [&](const std::monostate _) {
+                  return norm(
+                      a,
+                      std::holds_alternative<std::monostate>(axis)
+                          ? std::vector<int>()
+                          : get_reduce_axes(axis, a.ndim()),
+                      keepdims,
+                      stream);
+                }},
+            ord);
+      },
+      "a"_a,
+      "ord"_a = none,
+      "axis"_a = none,
+      "keepdims"_a = false,
+      "stream"_a = none,
+      R"pbdoc(
+    Matrix or vector norm.
+
+    This function is able to return matrix or vector norms,
+    depending on the value of the ``ord`` parameter.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array.  If `axis` is None, `a` must be 1-D or 2-D, unless `ord`
+        is None. If both `axis` and `ord` are None, the 2-norm of ``a.flatten`` will be returned.
+    ord : {non-zero int, inf, -inf, 'fro', 'nuc'}, optional
+        Order of the norm (see table under ``Notes``). inf means float(`inf`) object. The default is None.
+    axis : {None, int, 2-tuple of ints}, optional.
+        If `axis` is an integer, it specifies the axis of `a` along which to
+        compute the vector norms.  If `axis` is a 2-tuple, it specifies the
+        axes that hold 2-D matrices, and the matrix norms of these matrices
+        are computed.  If `axis` is None then either a vector norm (when `a`
+        is 1-D) or a matrix norm (when `a` is 2-D) is returned. The default
+        is None.
+    keepdims : bool, optional
+        If this is set to True, the axes which are normed over are left in the
+        result as dimensions with size one.  With this option the result will
+        broadcast correctly against the original `a`.
+
+    Returns
+    -------
+    n : array
+        Norm of the matrix or vector(s).
+
+    Notes
+    -----
+    For values of ``ord < 1``, the result is, strictly speaking, not a
+    mathematical 'norm', but it may still be useful for various numerical
+    purposes.
+
```

Comment:
This docstring should be in the same style as our other docstrings. See the docstrings in src/ops.cpp for example